### ISSUE TITLE
Enhancement: [no-unnecessary-type-arguments]  Check unnecessary type arguments in Tagged templates 

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#8706]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unnecessary-type-arguments

### Description

Currently, `no-unnecessary-type-arguments` does not report any errors when Tagged templates have unnecessary type arguments. 

see [playground](https://typescript-eslint.io/play#ts=5.4.2&showAST=es&fileType=.tsx&code=GYVwdgxgLglg9mABMOcA8AVRBeRYQC2ARgKYBOAfABQBQiiUJBADgDYCGjAzgFyIZM2nEgGUoZGGADmXAIJky7AJ4AaOonZ8Ma%2BkS00AlIgDe6gPRnEAOhs0AvjRop0%2BYuQoADACTGAjHZVEHwAmOw8AbhogA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWiIDmsALaIWqDJETRoHaJHBgAviF1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
```ts
function foo<T = number>(
  templates: TemplateStringsArray,
  a: T,
  b: T
) {
  // ...
}

foo<number>`${1}, ${2}`; 
```

I suggest that no-unnecessary-type-arguments also handles Tagged Templates.

### Fail

```TypeScript
function f<T = number>(templates: TemplateStringsArray, arg: T) {}
f<number>\`\${1}\`;
```

### Pass

```TypeScript
function f<T = number>(templates: TemplateStringsArray, arg: T) {}
f\`\${1}\`;
```
